isthmus/src/test/java/io/substrait/isthmus/NestedStructQueryTest.java

```diff
@@ -0,0 +1,427 @@
+import java.util.Arrays;
+import java.util.Map;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeFactory;
+import org.apache.calcite.schema.Schema;
+import org.apache.calcite.schema.Table;
+import org.apache.calcite.schema.impl.AbstractSchema;
+import org.apache.calcite.schema.impl.AbstractTable;
+import org.apache.calcite.sql.parser.SqlParseException;
+import org.apache.calcite.sql.type.SqlTypeName;
+import org.apache.calcite.util.Pair;
+import org.junit.jupiter.api.Test;
+
+public class NestedStructQueryTest extends PlanTestBase {
+  private class TypeHelper {
+    private final RelDataTypeFactory factory;
+
+    public TypeHelper(RelDataTypeFactory factory) {
+      this.factory = factory;
+    }
+
+    RelDataType struct(String field, RelDataType value) {
+      return factory.createStructType(Arrays.asList(Pair.of(field, value)));
+    }
+
+    RelDataType struct2(String field1, RelDataType value1, String field2, RelDataType value2) {
+      return factory.createStructType(
+          Arrays.asList(Pair.of(field1, value1), Pair.of(field2, value2)));
+    }
+
+    RelDataType struct3(
+        String field1,
+        RelDataType value1,
+        String field2,
+        RelDataType value2,
+        String field3,
+        RelDataType value3) {
+      return factory.createStructType(
+          Arrays.asList(Pair.of(field1, value1), Pair.of(field2, value2), Pair.of(field3, value3)));
+    }
+
+    RelDataType i32() {
+      return factory.createSqlType(SqlTypeName.INTEGER);
+    }
+
+    RelDataType string() {
+      return factory.createSqlType(SqlTypeName.VARCHAR);
+    }
+
+    RelDataType list(RelDataType elementType) {
+      return factory.createArrayType(elementType, -1);
+    }
+
+    RelDataType map(RelDataType key, RelDataType value) {
+      return factory.createMapType(key, value);
+    }
+  }
+
+  private void test(Table table, String query, String expectedExpressionText) {
+    final Schema schema =
+        new AbstractSchema() {
+          @Override
+          protected Map<String, Table> getTableMap() {
+            return Map.of("my_table", table);
+          }
+        };
+
+    try {
+      final SqlToSubstrait sqlToSubstrait = new SqlToSubstrait();
+      final SubstraitBuilder builder = new SubstraitBuilder(extensions);
+      Plan plan = sqlToSubstrait.execute(query, "nested", schema);
+      Expression obtainedExpression =
+          plan.getRelations(0).getRoot().getInput().getProject().getExpressions(0);
+      Expression expectedExpression = TextFormat.parse(expectedExpressionText, Expression.class);
+      assertEquals(expectedExpression, obtainedExpression);
+
+      ProtoPlanConverter converter = new ProtoPlanConverter();
+      io.substrait.plan.Plan plan2 = converter.from(plan);
+      assertPlanRoundrip(plan2);
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    } catch (SqlParseException e) {
+      throw new RuntimeException(e);
+    }
```

Comment:
Other tests using `assertPlanRoundtrip` don't try and catch the exceptions, rather they allow the methods/tests to throw. I think this is cleaner, because in practice the tests will fail if an exception is thrown.